src/daemon/rpmostreed-deployment-utils.c

```diff
@@ -229,7 +247,11 @@ rpmostreed_deployment_generate_variant (OstreeSysroot *sysroot,
       variant_add_commit_details (&dict, "base-", base_commit);
     }
   else
-    base_checksum = g_strdup (csum);
+    {
+      base_checksum = g_strdup (csum);
+      /* we would not consider endoflife attribute for layered commits */
```

Comment:
ah, I was thinking that, once the user has updated the commit to the newest, if end_of_life attribute is present, I am thinking that user should not be able to perform actions such as install/upgrade any more because the upstream branch stopped updating?

As a result, I think if we do not offer install/upgrade actions for that particular upstream branch, theoretically layered commit containing end-of-life attribute should not exist?( I am assuming that layered commits can only be achieved via install/upgrade actions, could be wrong tho)